test/acceptance/server-telemetry.bats

```diff
@@ -10,46 +10,29 @@ load _helpers
   kubectl create namespace acceptance
   kubectl config set-context --current --namespace=acceptance
 
+  # Install prometheus-operator and friends.
   helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
   helm repo update
-  helm install \
+  helm upgrade --install \
     --wait \
-    --version 39.6.0 \
+    --version 58.3.1 \
     prometheus prometheus-community/kube-prometheus-stack
 
-  helm install \
+  # Install Vault with telemetry config now that the prometheus CRDs are applied.
+  helm upgrade --install \
     --wait \
-    --values ./test/acceptance/server-test/telemetry.yaml \
+    --values ./test/acceptance/server-test/vault-server.yaml \
+    --values ./test/acceptance/server-test/vault-telemetry.yaml \
     "$(name_prefix)" .
 
-  wait_for_running $(name_prefix)-0
-
-  # Sealed, not initialized
-  wait_for_sealed_vault $(name_prefix)-0
-
-  # Vault Init
-  local token=$(kubectl exec -ti "$(name_prefix)-0" -- \
-    vault operator init -format=json -n 1 -t 1 | \
-    jq -r '.unseal_keys_b64[0]')
-  [ "${token}" != "" ]
-
-  # Vault Unseal
-  local pods=($(kubectl get pods --selector='app.kubernetes.io/name=vault' -o json | jq -r '.items[].metadata.name'))
-  for pod in "${pods[@]}"
-  do
-      kubectl exec -ti ${pod} -- vault operator unseal ${token}
-  done
-
   wait_for_ready "$(name_prefix)-0"
 
-  # Unsealed, initialized
-  local sealed_status=$(kubectl exec "$(name_prefix)-0" -- vault status -format=json |
-    jq -r '.sealed' )
-  [ "${sealed_status}" == "false" ]
+  echo 'path "sys/metrics" {capabilities = ["read"]}' | kubectl exec -i "$(name_prefix)-0" -- vault policy write metrics -
```

Comment:
Not sure if it's worth changing what you have there, but you could skip the need for the pipe, and passing STDIN, if you use process substitution.
```suggestion
  kubectl exec  "$(name_prefix)-0" -- sh -c "vault policy write metrics <(echo 'path \"sys/metrics\" {capabilities = [\"read\"]}')"
```

Reply:
I slightly prefer this how it is now, mainly for the fact that no quotes have to be escaped so I find it easier to read. That version would be better if we wanted e.g. Windows compatibility for the tests though, as it doesn't require the host to have a Unix-y shell.